tests/ui/doc_suspicious_footnotes.rs

```diff
@@ -0,0 +1,19 @@
+#![warn(clippy::doc_suspicious_footnotes)]
+
+/// This is not a footnote[^1].
+//~^ doc_suspicious_footnotes
+///
+/// This is not a footnote[^either], but it doesn't warn.
+///
+/// This is not a footnote\[^1], but it also doesn't warn.
+///
+/// This is not a footnote[^1\], but it also doesn't warn.
+///
+/// This is not a `footnote[^1]`, but it also doesn't warn.
+///
+/// This is a footnote[^2].
+///
+/// [^2]: hello world
```

Comment:
Could this also check for `#![doc]`, `//!`, `#[doc]` and the `include!`s macros?